gax-httpjson/src/test/java/com/google/api/gax/httpjson/ProtoRestSerializerTest.java

```diff
@@ -0,0 +1,141 @@
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package com.google.api.gax.httpjson;
+
+import com.google.common.truth.Truth;
+import com.google.protobuf.Field;
+import com.google.protobuf.Field.Cardinality;
+import com.google.protobuf.Option;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.Before;
+import org.junit.Test;
+
+public class ProtoRestSerializerTest {
+  private ProtoRestSerializer<Field> requestSerializer;
+  private Field field;
+  private String fieldJson;
+
+  @Before
+  public void setUp() {
+    requestSerializer = ProtoRestSerializer.create();
+    field =
+        Field.newBuilder()
+            .setNumber(2)
+            .setName("field_name1")
+            .addOptions(Option.newBuilder().setName("opt_name1").build())
+            .addOptions(Option.newBuilder().setName("opt_name2").build())
+            .setCardinality(Cardinality.CARDINALITY_OPTIONAL)
+            .build();
+
+    fieldJson =
+        "{\n"
+            + "  \"cardinality\": \"CARDINALITY_OPTIONAL\",\n"
+            + "  \"number\": 2,\n"
+            + "  \"name\": \"field_name1\",\n"
+            + "  \"options\": [{\n"
+            + "    \"name\": \"opt_name1\"\n"
+            + "  }, {\n"
+            + "    \"name\": \"opt_name2\"\n"
+            + "  }]\n"
+            + "}";
+  }
+
+  @Test
+  public void toJson() {
+    String fieldToJson = requestSerializer.toJson(field);
+    Truth.assertThat(fieldToJson).isEqualTo(fieldJson);
+  }
+
+  @Test
+  public void fromJson() {
+    Field fieldFromJson =
+        requestSerializer.fromJson(
+            new ByteArrayInputStream(fieldJson.getBytes(StandardCharsets.UTF_8)),
+            Field.newBuilder());
+
+    Truth.assertThat(fieldFromJson).isEqualTo(field);
+  }
+
+  @Test
+  public void fromJsonInvalidJson() {
+    try {
+      requestSerializer.fromJson(
+          new ByteArrayInputStream("heh".getBytes(StandardCharsets.UTF_8)), Field.newBuilder());
+      Truth.assertThat(true).isFalse();
```

Comment:
Is there a way to simply assert false, preferably with a message that parsing should have failed?

Reply:
Ditto on `assertThrows`.